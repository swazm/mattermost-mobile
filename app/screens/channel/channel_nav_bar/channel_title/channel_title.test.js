// Copyright (c) 2015-present SWAZM, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import ChannelTitle from './channel_title';

jest.mock('react-intl');

describe('ChannelTitle', () => {
    const baseProps = {
        displayName: 'My Channel',
        isGuest: false,
        hasGuests: false,
        canHaveSubtitle: false,
        isSelfDMChannel: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ChannelTitle {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot when isSelfDMChannel is true', () => {
        const props = {
            ...baseProps,
            displayName: 'My User',
            isSelfDMChannel: true,
        };
        const wrapper = shallow(
            <ChannelTitle {...props}/>,
            {context: {intl: {formatMessage: (intlId) => intlId.defaultMessage}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
