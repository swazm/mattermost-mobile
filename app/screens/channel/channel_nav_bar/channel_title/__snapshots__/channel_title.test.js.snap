// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelTitle should match snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "position": "relative",
        "top": -1,
        "width": "90%",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": undefined,
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      My Channel
    </Text>
    <Icon
      allowFontScaling={false}
      name="chevron-down"
      size={12}
      style={
        Object {
          "color": undefined,
          "marginHorizontal": 5,
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`ChannelTitle should match snapshot when isSelfDMChannel is true 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "position": "relative",
        "top": -1,
        "width": "90%",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": undefined,
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      <FormattedText
        defaultMessage="{displayName} (you)"
        id="channel_header.directchannel.you"
        values={
          Object {
            "displayName": "My User",
          }
        }
      />
    </Text>
    <Icon
      allowFontScaling={false}
      name="chevron-down"
      size={12}
      style={
        Object {
          "color": undefined,
          "marginHorizontal": 5,
        }
      }
    />
  </View>
</TouchableOpacity>
`;
